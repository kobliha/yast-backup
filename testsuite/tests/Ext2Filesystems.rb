# encoding: utf-8

# Backup module - testsuite
#
# Ext2Filesystems function tests
#
# testedfiles: backup/functions.ycp
#
# $Id$
#
module Yast
  class Ext2FilesystemsClient < Client
    def main
      Yast.import "UI"
      Yast.import "Testsuite"

      @readmap = {
        "target" => { "string" => "", "size" => -1, "tmpdir" => "/tmp" },
        "proc"   => { "cpuinfo" => { "value" => { "0" => { "flags" => "" } } } },
        "probe"  => {
          "has_smp" => false,
          "is_uml"  => false,
          "cdrom"   => { "manual" => [] }
        }
      }

      @execmap = {
        "target" => {
          "bash_output" => {
            "exit"   => 0,
            "stderr" => "",
            "stdout" => "charmap=\"ISO-8859-1\"\n"
          }
        }
      }

      Testsuite.Init([@readmap, {}, @execmap], nil)

      Yast.include self, "backup/functions.rb"

      @READ_mounts = {
        "proc" => {
          "mounts" => [
            {
              "file"    => "/",
              "freq"    => 0,
              "mntops"  => "rw",
              "passno"  => 0,
              "spec"    => "/dev/root",
              "vfstype" => "reiserfs"
            },
            {
              "file"    => "/proc",
              "freq"    => 0,
              "mntops"  => "rw",
              "passno"  => 0,
              "spec"    => "proc",
              "vfstype" => "proc"
            },
            {
              "file"    => "/dev/pts",
              "freq"    => 0,
              "mntops"  => "rw",
              "passno"  => 0,
              "spec"    => "devpts",
              "vfstype" => "devpts"
            },
            {
              "file"    => "/boot",
              "freq"    => 0,
              "mntops"  => "rw",
              "passno"  => 0,
              "spec"    => "/dev/hda1",
              "vfstype" => "ext2"
            },
            {
              "file"    => "/local",
              "freq"    => 0,
              "mntops"  => "rw",
              "passno"  => 0,
              "spec"    => "/dev/hda4",
              "vfstype" => "ext2"
            }
          ]
        },
        "etc"  => {
          "fstab" => [
            {
              "file"    => "/media/cdrom",
              "freq"    => 0,
              "mntops"  => "ro,noauto,user,exec",
              "passno"  => 0,
              "spec"    => "/dev/cdrom",
              "vfstype" => "auto"
            },
            {
              "file"    => "/unmountedk",
              "freq"    => 0,
              "mntops"  => "defaults",
              "passno"  => 0,
              "spec"    => "/dev/hdb1",
              "vfstype" => "ext2"
            }
          ]
        }
      }


      Testsuite.Test(lambda { Ext2Filesystems() }, [{}, {}, {}], nil)
      Testsuite.Test(lambda { Ext2Filesystems() }, [@READ_mounts, {}, {}], nil)

      nil
    end
  end
end

Yast::Ext2FilesystemsClient.new.main
