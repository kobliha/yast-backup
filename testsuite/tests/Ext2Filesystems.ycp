// Backup module - testsuite
//
// Ext2Filesystems function tests
//
// $Id$
//

{

include "backup/functions.ycp";
include "testsuite.ycp";

map READ_mounts = $[ "proc" : $[ "mounts" :
[
$["file":"/", "freq":0, "mntops":"rw", "passno":0,"spec":"/dev/root", "vfstype":"reiserfs"],
$["file":"/proc", "freq":0, "mntops":"rw", "passno":0, "spec":"proc", "vfstype":"proc"],
$["file":"/dev/pts", "freq":0, "mntops":"rw", "passno":0, "spec":"devpts", "vfstype":"devpts"],
$["file":"/boot", "freq":0, "mntops":"rw", "passno":0, "spec":"/dev/hda1", "vfstype":"ext2"],
$["file":"/local", "freq":0, "mntops":"rw", "passno":0, "spec":"/dev/hda4", "vfstype":"ext2"]
]],

"etc" : $[ "fstab" :
[
$["file":"/media/cdrom", "freq":0, "mntops":"ro,noauto,user,exec", "passno":0, "spec":"/dev/cdrom", "vfstype":"auto"],
$["file":"/unmountedk", "freq":0, "mntops":"defaults", "passno":0, "spec":"/dev/hdb1", "vfstype":"ext2"],
]

]];


TEST( ``(Ext2Filesystems()), [ $[], $[], $[] ], nil );
TEST( ``(Ext2Filesystems()), [ READ_mounts, $[], $[] ], nil );

}

