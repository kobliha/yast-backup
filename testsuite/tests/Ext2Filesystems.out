Read	.target.size "/etc/install.inf" -1
Read	.target.tmpdir "/tmp"
Dir	.autoyast2.desktop."s": []
Dir	.yast2.groups.s: []
Read	.probe.architecture nil
Read	.probe.has_apm nil
Read	.probe.has_pcmcia nil
Read	.probe.is_uml false
Read	.probe.has_smp false
Read	.probe.system nil
Read	.sysconfig.console.CONSOLE_ENCODING nil
Execute	.target.bash_output "locale -k charmap" $["exit":0, "stderr":"", "stdout":"charmap=\"ISO-8859-1\"\n"]
Read	.target.tmpdir "/tmp"
Read	.probe.floppy.manual nil
Read	.probe.cdrom.manual []
Read	.proc.mounts nil
Read	.etc.fstab nil
Return	[]
Read	.proc.mounts [$["file":"/", "freq":0, "mntops":"rw", "passno":0, "spec":"/dev/root", "vfstype":"reiserfs"], $["file":"/proc", "freq":0, "mntops":"rw", "passno":0, "spec":"proc", "vfstype":"proc"], $["file":"/dev/pts", "freq":0, "mntops":"rw", "passno":0, "spec":"devpts", "vfstype":"devpts"], $["file":"/boot", "freq":0, "mntops":"rw", "passno":0, "spec":"/dev/hda1", "vfstype":"ext2"], $["file":"/local", "freq":0, "mntops":"rw", "passno":0, "spec":"/dev/hda4", "vfstype":"ext2"]]
Read	.etc.fstab [$["file":"/media/cdrom", "freq":0, "mntops":"ro,noauto,user,exec", "passno":0, "spec":"/dev/cdrom", "vfstype":"auto"], $["file":"/unmountedk", "freq":0, "mntops":"defaults", "passno":0, "spec":"/dev/hdb1", "vfstype":"ext2"]]
Return	[$["mountpoint":"/boot", "partition":"/dev/hda1"], $["mountpoint":"/local", "partition":"/dev/hda4"], $["mountpoint":"/unmountedk", "partition":"/dev/hdb1"]]
